crypto/s2n_hmac.c

```diff
@@ -294,32 +311,41 @@ int s2n_hmac_digest_two_compression_rounds(struct s2n_hmac_state *state, void *o
 
 int s2n_hmac_free(struct s2n_hmac_state *state)
 {
+    if (!state) { return S2N_SUCCESS; }
```

Comment:
Is a non-null check enough? Don't we also need to check that `state` is writable?

Reply:
Since this is a "free" function, if `state` is NULL, there is nothing to be done.